### PR TITLE
[vanguard] Add property "description" to POI

### PR DESCRIPTION
**Pull request by peoplepeople000**
# Description
Add the property "description" to POI and make a input box to let user input the description of the reportData. Also, in the cluster drawer, we can see the description of each reportData.

# Changes
截圖 2024-03-20 晚上9 30 21
截圖 2024-03-20 晚上9 33 13

# Notes
Because there's no corresponding icon, so I used other icon instead.

# Checklist

- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any misspellings
- [x] Any changes to strings have been published to our translation tool
- [x] I conducted basic QA to assure all features are working
- [x] I requested code review from other team members

# Resolved Issues
#92

### CODE REVIEW

File: src/components/Drawer/AddReportDrawer/AddReportDrawerContent.tsx [hunk 1]
```diff
@@ -464,6 +372,45 @@ const StatusValueSelect: React.FC = () => {
...
+  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const newValue = e.target.value;
+    dispatch(
+      updateAddReportData({
+        target: {
+          ...reportData.target,
+          description: newValue,
+        },
+      }),
+    );
```
Comment (linyc0817) [inline src/components/Drawer/AddReportDrawer/AddReportDrawerContent.tsx:390]: It is better to pass the value directly. No need to create a new parameter.

File: src/components/Drawer/AddReportDrawer/AddReportDrawerContent.tsx [hunk 2]
```diff
@@ -604,7 +551,7 @@ const AddReportDrawerContent: React.FC = () => {
                 alt="status value"
```
Comment (linyc0817) [inline src/components/Drawer/AddReportDrawer/AddReportDrawerContent.tsx:551]: Alt means what this section means, so `status value` might not be suitable here.

File: src/components/Drawer/AddReportDrawer/AddReportDrawerContent.tsx [hunk 3]
```diff
@@ -604,7 +551,7 @@ const AddReportDrawerContent: React.FC = () => {
...
               />
             </div>
-            <StatusValueSelect />
+            <AddDescription />
```
Comment (linyc0817) [inline src/components/Drawer/AddReportDrawer/AddReportDrawerContent.tsx:554]: Follow the naming rules of previous components. e.g. StatusValueSelect
So you might use `StatusDescriptionAdd`? Or maybe you can come up with a better name.

File: src/components/Drawer/EditReportDrawer/EditReportDrawerContent.tsx [hunk 4]
```diff
@@ -69,6 +68,45 @@ const StatusValueSelect: React.FC = () => {
...
+  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const newValue = e.target.value;
+    dispatch(
+      updateAddReportData({
+        target: {
+          ...reportData.target,
+          description: newValue,
+        },
+      }),
+    );
```
Comment (linyc0817) [inline src/components/Drawer/EditReportDrawer/EditReportDrawerContent.tsx:86]: Same as above, pass value directly

File: src/components/Drawer/EditReportDrawer/EditReportDrawerContent.tsx [hunk 5]
```diff
@@ -199,6 +221,17 @@ const EditReportDrawerContent: React.FC = () => {
...
+        {/* report description */}
+        <div className="flex flex-row space-x-1 mt-1 items-center">
+          <div className="basis-0.5/12">
+            <Image
+              radius="none"
+              src={poiEditDrawerStatusValue}
+              alt="status value"
+            />
+          </div>
+          <EditDescription />
```
Comment (linyc0817) [inline src/components/Drawer/EditReportDrawer/EditReportDrawerContent.tsx:233]: Same as above, change the component name

**Review by linyc0817 — CHANGES_REQUESTED** [review]
Looks good overall.
Please help to check & fix some points.

File: src/components/Drawer/AddReportDrawer/AddReportDrawerContent.tsx (hunk 1 above)
Comment (peoplepeople000) [reply src/components/Drawer/AddReportDrawer/AddReportDrawerContent.tsx:390]: I removed the newValue parameter, and passed e.target.value to description directly.

File: src/components/Drawer/AddReportDrawer/AddReportDrawerContent.tsx (hunk 2 above)
Comment (peoplepeople000) [reply src/components/Drawer/AddReportDrawer/AddReportDrawerContent.tsx:551]: I changed the alt to "description".

File: src/components/Drawer/AddReportDrawer/AddReportDrawerContent.tsx (hunk 3 above)
Comment (peoplepeople000) [reply src/components/Drawer/AddReportDrawer/AddReportDrawerContent.tsx:554]: I changed the name of the component to StatusDescriptionAdd.

File: src/components/Drawer/EditReportDrawer/EditReportDrawerContent.tsx (hunk 4 above)
Comment (peoplepeople000) [reply src/components/Drawer/EditReportDrawer/EditReportDrawerContent.tsx:86]: I removed the newValue parameter, and passed e.target.value to description directly.

File: src/components/Drawer/EditReportDrawer/EditReportDrawerContent.tsx (hunk 5 above)
Comment (peoplepeople000) [reply src/components/Drawer/EditReportDrawer/EditReportDrawerContent.tsx:233]: I changed the name of the component to StatusDescriptionEdit.

**Review by linyc0817 — APPROVED** [review]
Well done.
But when i run the code, i found this warning `A component changed from controlled to uncontrolled.` at the console, can you try to fix it. (Seems like it may cause by NextUI )
截圖 2024-04-06 上午12 12 03

**Comment by linyc0817** [comment]
## Bug reproduce
https://github.com/CAMPUS-NYCU/smart-campus/assets/32840701/a4daea83-76e7-4b6c-a059-b74ab1d8f909

- May become error sometimes
截圖 2024-04-08 上午12 11 40

**Comment by peoplepeople000** [comment]
截圖 2024-04-17 晚上8 41 25
I added "" in row 101 to try to fix the bug.

截圖 2024-04-17 晚上8 43 37
I added "" in row 405 to try to fix the bug.

Because the error did not appear on my device. So I'm not pretty sure that if the error was fixed or not.

**Review by linyc0817 — APPROVED** [review]
Well done

**Pull request merged**